atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweeperServiceImpl.java

```diff
@@ -48,17 +56,36 @@ public SweepTableResponse sweepTable(
         SweepBatchConfig config = buildConfigWithOverrides(maxCellTsPairsToExamine, candidateBatchSize,
                 deleteBatchSize);
 
-        SweepResults sweepResults = fullSweep.orElse(true)
-                ? runFullSweepWithoutSavingResults(
-                        tableRef,
-                        decodedStartRow,
-                        config)
-                : runOneBatchWithoutSavingResults(
-                        tableRef,
-                        decodedStartRow,
-                        config);
-
-        return SweepTableResponse.from(sweepResults);
+        return SweepTableResponse.from(runSweep(fullSweep, tableRef, decodedStartRow, config));
+    }
+
+    private SweepResults runSweep(Optional<Boolean> fullSweep, TableReference tableRef, byte[] decodedStartRow,
+            SweepBatchConfig config) {
+        if (!fullSweep.isPresent()) {
```

Comment:
I don't think this will be ever triggered, as we have the `@Safe @QueryParam("fullSweep") @DefaultValue("true") Optional<Boolean> fullSweep` on the interface. Still, it doesn't seem harmless — the only drawback is that when changing the default, we probably want to change it in two places, which are not that far apart.

Reply:
This is irritating - there's no way to distinguish when the value was filled in by the web container.

Ho would you feel about removing the `@Default` annotation?  The default is still to true and documentation is unchanged, but then the defaulting no longer happens near where it is documented which is going to cause them to get out of sync.

Reply:
Do we need to log this at warn when we have the below info logs?

Reply:
I want to call attention to this if it happens - people shouldn't be using this endpoint in normal operation and there's a good chance it was called by mistake.